finagle-http/src/main/scala/com/twitter/finagle/http/codec/HttpServerDispatcher.scala

```diff
@@ -31,6 +31,11 @@ private[finagle] class HttpServerDispatcher(
     service.close()
   }
 
+  private def mayHaveContent(status: Status): Boolean = status match {
```

Comment:
> Sorry, its not clear to me if we agree. I believe at a Response with an empty body will get assigned a content length header of length 0. I haven't added tags to the handle function to check, but a raw Response should not be chunked and will not have a defined contentLength, so line 97 (on master, maybe this is the source of our confusion) should set the content-length to 0 on a 204 response without a body.

Yes, I understand the code in the same way. Thanks for your explanation! :)  (I was thinking about `Content-Length` of `rep` where the code on line 97 is _not_ applied.)

> That suggests to me (and we should get @mosesn's opinion in light of what we've found) that this isn't actually a job for a filter. This feels more like the things that are getting delegated to the netty pipeline such as RespondToExpectContinue. 

Okay, so you mean we create a `ChannelHandler`, like `RespondToExpectContinue`, instead of using a `Filter` and give it a responsibility to handle 204 and 304 responses? Correct? Honestly, I'm not familiar with Netty's pipeline and not quite sure how it works in finagle. What sort of high-level solution (or actual implementation) do you have in your mind? (Can you please be more specific?)